code/modules/reagents/chemistry/reagents/drugs.dm

```diff
@@ -858,4 +858,118 @@
 		update_flags |= M.adjustBruteLoss(rand(1,5)*REAGENTS_EFFECT_MULTIPLIER, FALSE)
 	return list(0, update_flags)
 
+
+//surge+, used in supercharge implants
+/datum/reagent/surge_plus
+	name = "Surge Plus"
+	id = "surge_plus"
+	description = "A superconducting gel that overloads processors, causing an effect reportedly similar to benzodiazepines in synthetic units."
+	reagent_state = LIQUID
+	color = "#28b581"
+
+	process_flags = SYNTHETIC
+	overdose_threshold = 30
+	addiction_chance = 1
+	addiction_chance_additional = 20
+	addiction_threshold = 5
+	addiction_decay_rate = 0.2
+	taste_description = "silicon"
+
+
+/datum/reagent/surge_plus/on_mob_life(mob/living/M)
+	var/update_flags = STATUS_UPDATE_NONE
+	M.AdjustParalysis(-8 SECONDS)
+	M.AdjustStunned(-8 SECONDS)
+	M.AdjustWeakened(-8 SECONDS)
+	M.AdjustKnockDown(-8 SECONDS)
+	update_flags |= M.adjustStaminaLoss(-25, FALSE)
+	if(prob(5))
+		var/high_message = pick("You feel calm.", "You feel collected.", "You feel like you need to relax.")
+		if(prob(10))
+			high_message = "0100011101001111010101000101010001000001010001110100111101000110010000010101001101010100!"
+		to_chat(M, "<span class='notice'>[high_message]</span>")
+
+
+	return ..() | update_flags
+
+/datum/reagent/surge_plus/overdose_process(mob/living/M, severity)
+	var/update_flags = STATUS_UPDATE_NONE
+	var/recent_consumption = holder.addiction_threshold_accumulated[type]
+	M.Jitter(40 SECONDS)
+	M.Stuttering(10 SECONDS)
+	if(prob(5 * DRAWBACK_CHANCE_MODIFIER(recent_consumption)))
+		to_chat(M, "<span class='notice'>Your circuits overheat!</span>") // synth fever
+		M.bodytemperature += 30 * recent_consumption
+		M.Confused(2 SECONDS * recent_consumption)
+	if(prob(10))
+		to_chat(M, "<span class='danger'>You experience a violent electrical discharge!</span>")
+		playsound(get_turf(M), 'sound/effects/eleczap.ogg', 75, 1)
+		//Lightning effect for electrical discharge visualization
+		var/icon/I=new('icons/obj/zap.dmi',"lightningend")
+		I.Turn(-135)
+		var/obj/effect/overlay/beam/B = new(get_turf(M))
+		B.pixel_x = rand(-20, 0)
+		B.pixel_y = rand(-20, 0)
+		B.icon = I
+		update_flags |= M.adjustFireLoss(rand(1,5)*REAGENTS_EFFECT_MULTIPLIER, FALSE)
+		update_flags |= M.adjustBruteLoss(rand(1,5)*REAGENTS_EFFECT_MULTIPLIER, FALSE)
+	return list(0, update_flags)
+
+//Servo Lube, supercharge
+/datum/reagent/lube/combat
```

Comment:
As before, I'd recommend making this a subtype of ultralube and changing what you need, as a lot of this is copy-pasted from there.